code/modules/projectiles/guns/energy/kinetic_accelerator.dm

```diff
@@ -218,8 +218,8 @@
 		var/turf/closed/mineral/M = target_turf
 		M.gets_drilled(firer, TRUE)
 		if(iscarbon(firer))
-			var/mob/living/carbon/C = firer
-			var/skill_modifier = C?.mind.get_skill_modifier(/datum/skill/mining, SKILL_SPEED_MODIFIER)
+			var/mob/living/carbon/carbon_firer = firer
+			var/skill_modifier = carbon_firer.mind.get_skill_modifier(/datum/skill/mining, SKILL_SPEED_MODIFIER)
```

Comment:
You don't check for mind here ?

Reply:
Added new commit checking for mind in a separate if statement so that skill_modifier doesn't become 0.